lib/repositories/firehose-repository.ts

```diff
@@ -0,0 +1,53 @@
+import { Firehose } from 'aws-sdk';
+import { AnalyticsEvent, AnalyticsEventType } from '../entities/analytics-events';
+import { default as Logger } from 'bunyan';
+
+export class FirehoseLogger {
+  private log: Logger;
+  private readonly streamName: string;
+  private readonly firehose: Firehose;
+  private readonly isMock: boolean;
+
+  constructor(_log: Logger, streamArn: string, isMock: boolean = false) {
+    this.log = _log;
+    // Split the streamArn to extract the streamName
+    const streamArnParts = streamArn.split('/');
+    if (streamArnParts.length !== 2) {
+      throw new Error('Invalid ANALYTICS_STREAM_ARN environment variable');
+    }
+    this.streamName = streamArnParts[1];
+    this.firehose = new Firehose();
+    this.isMock = isMock;
+  }
+ 
+  async sendAnalyticsEvent(
+    eventData: {
+      eventType: AnalyticsEventType;
+      eventProperties: { [key: string]: any };
+    }
+  ): Promise<{ statusCode: number; body: string }> {
+    const { eventType, eventProperties } = eventData;
+    const analyticsEvent = new AnalyticsEvent(eventType, eventProperties);
+
+    if (this.isMock) {
+      return { statusCode: 200, body: 'Record put successful in mock' };
+    }
```

Comment:
nit: would it be possible to remove this and just mock the return value of `this.firehose.putRecord` when we are testing this? if that makes it too tricky, can ignore this comment